packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -94,6 +88,18 @@ export async function preResolveOptions(
 	return merged;
 }
 
+function mergeConfigs<T>(...configs: T[]): PreResolvedOptions {
```

Comment:
our options are getting deeply nested structures. to avoid manual ...expansion, use deepmerge.

This one is a bit tricky, maybe warrants extra tests.